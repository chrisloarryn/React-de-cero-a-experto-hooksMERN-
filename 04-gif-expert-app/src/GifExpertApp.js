import React, { useState } from 'react'

import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'
// import PropTypes from 'prop-types'

export const GifExpertApp = ( { defaultCategories = [] } ) => {
  const [categories, setCategories] = useState(defaultCategories) // [{ name: 'One Piece' }]

  //   const handleAdd = () => {
  //     // setCategories([...categories, {name: 'Ranma'}])
  //     setCategories((cats) => [...cats, {name: 'Ranma'}])
  //   }

  return (
    <div>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map(({ name }, index) => (
          <GifGrid key={name} category={name} />
        ))}
      </ol>
    </div>
  )
}

// GifExpertApp.propTypes = {}
