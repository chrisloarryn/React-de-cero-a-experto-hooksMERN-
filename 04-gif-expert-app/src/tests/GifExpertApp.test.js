import React from 'react'
import { shallow } from 'enzyme'
import { GifExpertApp } from './../GifExpertApp'

import '@testing-library/jest-dom'

// import { useFetchGifs } from '../../hooks/useFetchGifs'
// jest.mock('../../hooks/useFetchGifs')

describe(`Test into <GifExpertApp /> component`, () => {
  test('test into the component', () => {
    const wrapper = shallow(<GifExpertApp />)
    expect(wrapper).toMatchSnapshot()
  })

  test(`should show a list of categories`, () => {
    const categories = [
      {
        name: 'One Piece'
      },
      {
        name: 'Dragon Ball'
      }
    ]
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categories.length)
  })
})
