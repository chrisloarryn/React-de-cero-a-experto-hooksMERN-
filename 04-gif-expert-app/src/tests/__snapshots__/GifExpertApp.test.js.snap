// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test into <GifExpertApp /> component should show a list of categories 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;

exports[`Test into <GifExpertApp /> component test into the component 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
